daemon/mgr/container_exec.go

```diff
@@ -130,7 +134,9 @@ func (mgr *ContainerManager) StartExec(ctx context.Context, execid string, attac
 		return err
 	}
 
+	execConfig.Lock()
 	execConfig.Running = true
+	execConfig.Unlock()
 
 	err = mgr.Client.ExecContainer(ctx, &ctrd.Process{
 		ContainerID: execConfig.ContainerID,
```

Comment:
Overall, I think lock for writing is reasonable.

Reply:
I think the most important thing is that whether the added lock could satisfy the business demand. 

Lock is used to lock the race section. If we only lock race section when writing. The it will lead to : 
* multiple write could go in mutex exclusion;
* multiple read will not be exclusive;
* write and read would be in chaos. Lock on writes does nothing when race section is being read by other routines.

Just confirm if we could satisfy the demand. @Ace-Tang